### DISCUSSION TITLE
Can iOS read multiple tags without dismissing its UI?

### DISCUSSION THREAD

**Discussion by dcanora** [revtel/react-native-nfc-manager#720] [Q&A] [answered]
I'm trying to create a ticketing-type app. These are not NDEF enconded, and I'm using `.sendMifareCommandIOS` to read the tags. I'd like to read multiple tags in succession, but on iOS that doesn't seem possible. The only way I can make it work is if the iOS NFC UI is dismissed after each read and the user hits my Start Scanning button again to read the next tag.

Am I missing something or is this a limitation on iOS?

**Comment by pke** [comment]
in iOS there is the `invalidateAfterFirstRead` flag in the native code, which allows to read multiple tags

https://developer.apple.com/documentation/corenfc/nfcndefreadersession/2882064-init

Not sure this is exposed by this lib

**Reply by dcanora** [reply]
Great info, thank you @pke!   

Anyone know if this is supported in the lib?

**Reply by nik27** [reply]
It is exposed but I've never used it so I don't know if it works as expected.

It's part of `RegisterTagEventOpts` and can be used in `requestTechnology` function as prop of `options` object (second argument). Like this:

```javascript
await nfcManager.requestTechnology(NfcTech.Ndef, {invalidateAfterFirstRead: false})
```

**Reply by dcanora** [reply, marked as answer]
Thanks, @nik27!  I was able to use `NfcManager.restartTechnologyRequestIOS()` for subsequent reads, which is working nicely. Cheers.
